miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -389,16 +387,13 @@ private void validateAliasUniqueness(Sample sample) throws ConstraintViolationEx
   public void confirmExternalNameUniqueForProjectIfRequired(String newExternalName, Sample sample)
       throws IOException, ConstraintViolationException {
     if (!isUniqueExternalNameWithinProjectRequired()) return;
-    for (SampleIdentity existingIdentity : getIdentitiesByExternalNameOrAlias(newExternalName)) {
-      // not an issue if it matches an identity from another project
-      if (existingIdentity.getProject().getId() != sample.getProject().getId()) continue;
-      Set<String> intersection = new HashSet<>(SampleIdentityImpl.getSetFromString(newExternalName));
-      intersection.retainAll(SampleIdentityImpl.getSetFromString(existingIdentity.getExternalName()));
-      if (intersection.size() != 0) {
+    Collection<SampleIdentity> matches = getIdentitiesByExactExternalNameAndProject(newExternalName,
+        sample.getProject().getId());
+    if(matches.size() != 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)